bundles/org.openhab.binding.xmppclient/src/main/java/org/openhab/binding/xmppclient/internal/client/XMPPClient.java

```diff
@@ -73,14 +78,20 @@ public void unsubscribe(XMPPClientMessageSubscriber channel) {
         subscribers.remove(channel);
     }
 
-    public void connect(String host, Integer port, String login, String domain, String password,
+    public void connect(String host, Integer port, String login, String nick, String domain, String password,
             SecurityMode securityMode) throws XMPPClientConfigException, XMPPClientException {
         disconnect();
         String serverHost = domain;
         if (!host.isBlank()) {
             serverHost = host;
         }
 
+        if (!nick.isEmpty()) {
```

Comment:
Guess you don't want whitespace names
```suggestion
        if (!nick.isBlank()) {
```

Reply:
good point, updated.